configurations/configValidators.go

```diff
@@ -0,0 +1,62 @@
+package configurations
+
+import (
+	"assignment-exec/image-builder/utilities/validation"
+	"github.com/pkg/errors"
+)
+
+type configValidator func(AssignmentEnvConfig) error
+
+func ValidatorForConfig(d AssignmentEnvConfig, configValidators ...configValidator) validation.Validator {
+	return func() error {
+		for _, cv := range configValidators {
+			if err := cv(d); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+}
+
+func withBaseImageValidator() configValidator {
+	return func(d AssignmentEnvConfig) error {
+		// Base Image name cannot be empty string.
+		if d.BaseImage == "" {
+			return errors.New("base image name cannot be empty string")
+		}
+
+		err := validateBaseImage(d.BaseImage)
+		if err != nil {
+			return err
+		}
+		return nil
```

Comment:
Line 28-32 can be replaced with `return validateBaseImage(c.BaseImage)`

Reply:
Done